apple/RNGestureHandlerButton.h

```diff
@@ -19,4 +26,13 @@
 @property (nonatomic, assign) UIEdgeInsets hitTestEdgeInsets;
 @property (nonatomic) BOOL userEnabled;
 
+#if TARGET_OS_OSX
+#ifdef __cplusplus
+- (void)mountChildComponentView:(RNGHUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index;
+- (void)unmountChildComponentView:(RNGHUIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index;
+- (void)updateLayoutMetrics:(const facebook::react::LayoutMetrics &)layoutMetrics
+           oldLayoutMetrics:(const facebook::react::LayoutMetrics &)oldLayoutMetrics;
+#endif
```

Comment:
I actually have no idea whether it's safe to do (have method declarations behind a compile flag but not the implementation.

Why do it this way instead of just changing files where it was included to be Objective-C++?

Reply:
This file was already change into `Obj-C++`. The thing is, changing it to `Obj-C++` is not enough and without this flag you'll get compilation errors. We talked about it with @piaskowyk and this was the solution that he came up with. Let us know if you see better one 😅

Reply:
Well, that makes sense.

Done in [9ebcbff](https://github.com/software-mansion/react-native-gesture-handler/pull/3154/commits/9ebcbff181ec0260c78581579bbd4885d8983bd5)

Reply:
Checks for `__cplusplus` shouldn't be necessary in that case

Reply:
You're right, done in [d8d1772](https://github.com/software-mansion/react-native-gesture-handler/pull/3154/commits/d8d177295dced6a4e4e94696df2699f73d1faf19)